### ISSUE TITLE
Typescript interface is wrong

### ISSUE THREAD

**Issue by swftvsn** [image-size/image-size#218]
For an example in ico.ts if I add types, one can see that the return value does not conform to ISizes:

```ts
calculate(buffer): ISizes | ISize {
    const nbImages = buffer.readUInt16LE(4)
    const imageSize: ISize = getImageSize(buffer, 0)

    if (nbImages === 1) {
      return imageSize
    }

    const result: ISizes = {
      height: imageSize.height,
      images: [imageSize],
      width: imageSize.width
    }

    for (let imageIndex = 1; imageIndex < nbImages; imageIndex += 1) {
      result.images.push(getImageSize(buffer, imageIndex))
    }

    return result
  }
```

Either the ISIzes type is wrong or the return value is wrong. Which one is it?

ISizes is currently declared like
```ts
export interface ISizes {
  result: ISize[]
  type?: string
}
```

But all the functions that I've seen actually return 
```ts
export interface ISizes {
  images: ISize[]
  height: number
  width: number
}
```
Where the height and width are from the first image. So, you could probably declare the return value as:
```ts
export interface ISize {
  width: number | undefined
  height: number | undefined
  orientation?: number
  type?: string
}

export interface ISizeCalculationResult extends ISize {
  images?: ISize[]
}

export interface IImage {
  validate: (buffer: Buffer) => boolean
  calculate: (buffer: Buffer, filepath?: string) => ISizeCalculationResult
}
```
Which reflects the real world as of now.

**Comment by swftvsn** [comment]
Added PR #219 to fix this.

**Comment by swftvsn** [comment]
@netroy Can you please review the changes, merge the PR + release a new version? Thank you!

**Comment by netroy** [comment]
@swftvsn published as `0.8.3`
